website/docs/r/capacity_reservation.html.markdown

```diff
@@ -0,0 +1,86 @@
+---
+subcategory: "Compute"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_capacity_reservation"
+description: |-
+  Manages a Capacity Reservation within a Capacity Reservation Group.
+---
+
+# azurerm_capacity_reservation
+
+Manages a Capacity Reservation within a Capacity Reservation Group.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "example-rg"
+  location = "West Europe"
+}
+
+resource "azurerm_capacity_reservation_group" "example" {
+  name                = "example-capacity-reservation-group"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+}
+
+resource "azurerm_capacity_reservation" "example" {
+  name                            = "example-capacity-reservation"
+  resource_group_name             = azurerm_resource_group.example.name
+  location                        = azurerm_resource_group.example.location
+  capacity_reservation_group_name = azurerm_capacity_reservation_group.example.name
+  sku {
+    name     = "Standard_D2s_v3"
+    capacity = 1
+  }
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Specifies the name of this Capacity Reservation. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) Specifies the name of the resource group the Capacity Reservation is located in. Changing this forces a new resource to be created.
+
+* `location` - (Required) The Azure location where the Capacity Reservation exists. Changing this forces a new resource to be created.
+
+* `capacity_reservation_group_name` - (Required) The name of the Capacity Reservation Group where the Capacity Reservation exists. Changing this forces a new resource to be created.
+
+* `sku` - (Required) A `sku` block as defined below.
+
+* `zone` - (Optional) Specifies a list of Availability Zone which this Capacity Reservation reserves. Changing this forces a new resource to be created.
+
+* `tags` - (Optional) A mapping of tags to assign to the resource.
+
+---
+
+An `sku` block exports the following:
+
+* `name` - (Required) Name of the sku.Changing this forces a new resource to be created.
+
+* `capacity` - (Required) Specifies the number of instances to be reserved.
```

Comment:
There is no fixed values for them, and both are required to be specified by user and not decided by API.

name is the size of vm, e.g. `Standard_D2s_v3`, since the type keeps growing, I set the validation to be `StringIsNotEmpty`, like what we did or VM resource https://github.com/hashicorp/terraform-provider-azurerm/blob/7c27c458db592eb8ffc47b2fe646f0b00f399f48/internal/services/compute/linux_virtual_machine_resource.go#L83-L86
capacity can be set up to the quota of the v-cores in the subscription, so didn't add the limitation. the validation function is set to `IntAtLeast(1)`